components/motor/gpiostepper/gpiostepper.go

```diff
@@ -226,37 +211,39 @@ func (m *gpioStepper) doCycle(ctx context.Context) (time.Duration, error) {
 	m.lock.Lock()
 	defer m.lock.Unlock()
 
+	// thread waits until something changes the target posiiton in the
+	// gpiostepper struct
 	if m.stepPosition == m.targetStepPosition {
 		return 5 * time.Millisecond, nil
 	}
 
+	// TODO: Setting PWM here works much better than steps to set speed
+	// Redo this part with PWM logic.
 	err := m.doStep(ctx, m.stepPosition < m.targetStepPosition)
 	if err != nil {
 		return time.Second, fmt.Errorf("error stepping %w", err)
 	}
 
-	return time.Duration(int64(time.Microsecond*1000*1000) / int64(math.Abs(float64(m.targetStepsPerSecond)))), nil
```

Comment:
An added delay to the stepper delay which was sometimes interrupted by the next tick, a negative number which mean no delay at all, or a wrong delay since we weren't accounting for the rpm -> rps -> microsecond/nano/second conversions.